pkg/adapter/nimbus-kubearmor/processor/kspbuilder.go

```diff
@@ -39,6 +40,19 @@ func BuildKspsFrom(logger logr.Logger, np *v1.NimbusPolicy) []kubearmorv1.KubeAr
 	return ksps
 }
 
+func setDefaultValues(ksp *kubearmorv1.KubeArmorPolicy) {
+	if len(ksp.Spec.Network.MatchProtocols) == 0 {
+		ksp.Spec.Network.MatchProtocols = append(ksp.Spec.Network.MatchProtocols, kubearmorv1.MatchNetworkProtocolType{
+			Protocol: "raw",
+		})
+	}
+	if len(ksp.Spec.Capabilities.MatchCapabilities) == 0 {
+		ksp.Spec.Capabilities.MatchCapabilities = append(ksp.Spec.Capabilities.MatchCapabilities, kubearmorv1.MatchCapabilitiesType{
+			Capability: "lease",
+		})
+	}
+}
```

Comment:
Don't set default values, there is already a [PR](https://github.com/kubearmor/KubeArmor/pull/1547) in place to omit the empty fields in KubeArmor.